### PR TITLE
Accessibility Improvements (webpack)

### PR DESCRIPTION
**Pull request by crhallberg**
Chris and Geoff are evaluating the accessibility of the UV.

## Contrast
- [x] Page dropdown contrast (bg #10819A, alternatively: text color to #2F2F2F or darker)
- [x] Go button contrast (bg #10819A, alternatively: text color to #2F2F2F or darker)
- [x] Link contrast (#10819A)
- [x] Disabled button contrast

## Markup issues
- [x] Zoom button labels
- [x] Font too small (`hide-text() { font: 0/0 a; }`)
  - [x] Fix broken image buttons
- [x] Redundant titles
- [x] [FooterPanel buttons](#discussion_r455865264)
- [x] Search within label (1.1.1) (#750)
  - [ ] Search for `.label` elements

## Screenreader Enhancements
- [x] Add jump link to downloads (alt versions) (1.2.x)
- [x] Add heading above bottom menu
- [x] Improve heading hierarchy (#750)

## Keyboard Support
- [x] Focus on whole iframe likely confusing (`tabindex="-1"`)
- [ ] Navigate nav buttons with arrow keys
- [x] Dropdown doesn't close on unfocus (#457)
- [ ] Cannot tab to page thumbnails
  - [x] Turn div into button or add `tabindex="0"`
  - [x] On select, focus on center panel
  - [ ] On escape, go back to thumbnail (ideal)
- [ ] SeaDragon UI fades out even when UI elements focused (might be out of scope)
- [x] PageUp/Down is good, is it expected behavior?
- [x] `<a href="/" id="top"/>`

### CODE REVIEW

File: src/modules/uv-pdfheaderpanel-module/css/styles.less [hunk 1]
```diff
@@ -7,7 +7,6 @@
         color: @text-secondary-color;
 
         button.btn.imageBtn {
-            font-size: 0;
             padding-left: 2px;
```
Comment (stephenwf) [inline src/modules/uv-pdfheaderpanel-module/css/styles.less:10]: Noticing the text appearing without the screen reader, can we use `.sr-only();` here too?

File: src/modules/uv-searchfooterpanel-module/FooterPanel.ts [hunk 2]
```diff
@@ -132,11 +132,7 @@ export class FooterPanel extends BaseFooterPanel {
     this.$searchPagerContainer.prepend(this.$searchPagerControls);
 
     this.$previousResultButton = $(
-      '<a class="previousResult" title="' +
-        this.content.previousResult +
-        '">' +
-        this.content.previousResult +
-        "</a>"
+      '<a class="previousResult">' + this.content.previousResult + "</a>"
```
Comment (stephenwf) [inline src/modules/uv-searchfooterpanel-module/FooterPanel.ts:135]: As these don't have `href` attributes, should we make them buttons?

File: src/modules/uv-pdfheaderpanel-module/css/styles.less (hunk 1 above)
Comment (stephenwf) [reply src/modules/uv-pdfheaderpanel-module/css/styles.less:10]: Screenshot 2020-07-16 at 16 16 27
For reference from the Netlify ^

File: src/modules/uv-searchfooterpanel-module/FooterPanel.ts (hunk 2 above)
Comment (crhallberg) [reply src/modules/uv-searchfooterpanel-module/FooterPanel.ts:135]: Yes!

File: src/modules/uv-pdfheaderpanel-module/css/styles.less (hunk 1 above)
Comment (crhallberg) [reply src/modules/uv-pdfheaderpanel-module/css/styles.less:10]: This is an inadvertently added bug. The `font-size: 0` is probably a fine if unconventional solution for hiding text.

**Comment by crhallberg** [comment]
At this point, all markup issues are being treated as critical and all the JS issues are determined to be nice to have, depending on the deadline (Sept 17) and how difficult they are to implement.

**Comment by edsilv** [comment]
Can we change the link colour back to #14a4c3? This passes AA

**Comment by crhallberg** [comment]
Wrapping up for the day and I wanted feedback on my thoughts.

- Once you've tabbed into the content area, you need a way out. [This is likely solved by good headings](https://webaim.org/projects/screenreadersurvey8/#finding).
- Once you've selected and viewed a page and you want to go back to the thumbnails, should we add a shortcut (esc?). This is also likely solved by good headings.
- Should we add arrow key support to the thumbnails? Definitely nice to have.

Discussion of headers: https://github.com/UniversalViewer/universalviewer/issues/750#issuecomment-690619676

**Comment by demiankatz** [comment]
@crhallberg, given the timeline, I think anything that is not essential should probably be deferred until the "must haves" are completed. Any "nice to haves" are worth capturing as issues with a tag of accessibility so we can tackle them in the near future. Some more specific thoughts:

- Regarding a "way out" of the content area, are you suggesting that once you tab in, you can't tab back out again?

- Regarding a shortcut back to the thumbnails, I think the main thing to watch out for is that, with all of the UV's different modules, selecting a thumbnail can potentially trigger a variety of different viewers to load in the content panel, and each of those viewers may have different keyboard controls of its own; we should be careful not to create a conflict or inconsistency. (It's been a while, so I can't remember how much of the thumbnail navigation code is shared between extensions, and how much is shared... but whether or not it's the same code, it has the same basic structure, so making it behave consistently is important).

- Regarding thumbnail arrow key support, I agree it's nice to have, but that's one I'd make an issue for and defer until later, since I don't think accessibility is completely lost without it.

**Comment by crhallberg** [comment]
I'm on vacation all next week so I wanted to give a clear update as to what's left in my head.

### This Week

- [ ] Skip to download
- [ ] Metadata link contrast
- [ ] Add header above bottom menu
- [ ] Search for remaining non-label `.label`s

### Needed by Sept 30

- [ ] Check internationalization

### Nice to have

- [ ] Arrow key support
  - [ ] Header nav
  - [ ] Thumbnails
- [ ] Match thumbnail focus to page up/down controls

**Comment by edsilv** [comment]
superseded by #767

**Pull request closed without merging**